diff --git a/Pipeto/src/commands/activate_emergency_protocols.c b/Pipeto/src/commands/activate_emergency_protocols.c
index dffbdfc..42b1560 100644
--- a/Pipeto/src/commands/activate_emergency_protocols.c
+++ b/Pipeto/src/commands/activate_emergency_protocols.c
@@ -11,20 +11,20 @@
 #include <stdbool.h>
 #include "pipeto.h"
 
-char *hash(char *key)
+char *hash(char *str)
 {
-    key = strdup(key);
-    for (int i = 0; key[i] != '\0'; i++)
-        key[i] = key[i] + i - strlen(key);
-    return key;
+    str = strdup(str);
+    for (int i = 0; str[i] != '\0'; i++)
+        str[i] = str[i] + i - strlen(str);
+    return str;
 }
 
 void activate_emergency_protocols()
 {
     char input[100];
-    char *pass = getenv("EMERGENCY_PASSWD");
+    char *password = getenv("EMERGENCY_PASSWD");
 
-    if (pass == NULL)
+    if (password == NULL || !password)
         return;
     printf("Enter emergency password: ");
     fgets(input, sizeof(input), stdin);
@@ -34,7 +34,7 @@ void activate_emergency_protocols()
         printf(" protocols not activated.\n");
         return;
     }
-    if (strcmp(hash(input), hash(pass)) == 0) {
+    if (strcmp(hash(input), hash(password)) == 0) {
         printf("{Emergency protocols activated, you are now admin !}\n");
         IS_ADMIN = true;
     }
